fixtures/load-repo.js

```diff
@@ -0,0 +1,205 @@
+  });
+
+  const filesWithNumNodeTypes = await Promise.all(
+    interestingFiles.map(async ({ file, contents }) => {
+      const seenNodeTypes = new Set();
+      let numNodes = 0;
+      let parsed;
+      /**
+       * This should be returned if the file is invalid
+       * This could happen if there is a parsing error or if it is empty
+       * or if it has too many linting errors, etc
+       */
+      const invalidResult = {
+        file,
+        contents,
+        score: 0,
+        numLintErrors: 0,
+        numNodeTypes: 0,
+        loc: 0,
+      };
+      const filenameRelative = path.join(reposDir, file);
+      try {
+        // Attempts to parse each file with babel
+        // Traverses the file and counts up the number of node types and number of nodes
+        // Files with more node types tend to be more interesting and better candidates for lint testing
+        parsed = await babel.parseAsync(contents, {
+          parserOpts: {
+            plugins: [
+              'jsx',
+              'classProperties',
+              'dynamicImport',
+              'nullishCoalescingOperator',
+              'optionalChaining',
+            ],
+          },
+          filenameRelative,
+        });
+      } catch (error) {
+        return invalidResult;
+      }
+
+      babel.traverse(parsed, {
+        enter(path) {
+          numNodes++;
+          seenNodeTypes.add(path.type);
+        },
+      });
+
+      const numNodeTypes = seenNodeTypes.size;
+
+      let eslintReport;
+      try {
+        eslintReport = eslint.executeOnText(contents).results[0];
+      } catch (error) {
+        console.log('eslint error', file, error);
+        return invalidResult;
+      }
+
+      const numLintErrors = eslintReport.errorCount + eslintReport.warningCount;
+      const eslintFixedContents = eslintReport.output || contents;
+
+      let prettifiedContents;
+
+      try {
+        prettifiedContents = prettier.format(eslintFixedContents, prettierOpts);
+      } catch (error) {
+        console.log('prettier error', file, error);
+        return invalidResult;
+      }
+
+      // Generates a "score" to a file based on the # of nodes, # of node types
+      // Used for sorting
+      // NumNodeTypes is weighted much more strongly
+      // because files tend to have a lot of nodes compared to the number of node types
+      const score = numNodeTypes * 10 + numNodes / 1000 - numLintErrors * 30;
+      return {
+        file,
+        contents: prettifiedContents,
+        score,
+        numLintErrors,
+        numNodeTypes,
+        loc: contents.split('\n').length,
+      };
+    })
+  );
+
+  const sortedFiles = filesWithNumNodeTypes
+    // Make sure it parsed correctly, has a reasonable number of errors, and is not too short
+    .filter(
+      (file) => file.score > 0 && file.numLintErrors < 100 && file.loc > 50
+    )
+    .sort((a, b) => b.score - a.score)
+    .slice(0, 50);
+
+  const { chosenFiles } = await prompts({
```

Comment:
I wish I had known about `prompts` the last time I tried to do something like this.

Reply:
Yeah it's a super nice library! I used to use [`inquirer`](https://github.com/SBoudrias/Inquirer.js/) for this kind of thing (it's a much more popular library), but `prompts` is nice since it is a much smaller library and has a lot fewer dependencies (because of [terkelg](https://github.com/terkelg))